sdks/python/apache_beam/io/iobase.py

```diff
@@ -1487,10 +1497,10 @@ def split(self, element, restriction):
       # Invoke source.split to get initial splitting results.
       source_bundles = self._source.split(self._desired_chunk_size)
       for source_bundle in source_bundles:
-        yield source_bundle
+        yield (source_bundle, None)
 
     def restriction_size(self, element, restriction):
-      return restriction.weight
+      return restriction[0].weight
 
     def restriction_coder(self):
       return coders.DillCoder()
```

Comment:
Do we ever encode the result of current_restriction()? (For example, due to self-checkpointing?)

Reply:
Thinking about this some more, I think it'd be safer and cleaner to create an actual _SDFBoundedSourceRestriction class that holds the SourceBundle and the (lazily-initialized) underlying restriction rather than using a tuple. Its `__reduce__` method could return only the former, and its constructor only take the former. Many of the methods here would simply delegate to it.

Reply:
> Do we ever encode the result of current_restriction()? (For example, due to self-checkpointing?)

I would say, no. `current_restriction` is expected be called inside SDF.DoFn(). Split(include self-checkpoint) should go through `try_split`.

But I really like the `_SDFBoundedSourceRestriction`, definitely a better way to go. Thanks for your help!